### PR TITLE
fix vsphere TLS client config overwrite

### PR DESCRIPTION
**Pull request by mate4st**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** 
Possible Fix for #6859

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/provider/cloud/vsphere/provider.go [hunk 1]
```diff
@@ -91,9 +90,7 @@ func newSession(ctx context.Context, dc *kubermaticv1.DatacenterSpecVSphere, use
 	// creating the govmoni Client in roundabout way because we need to set the proper CA bundle: reference https://github.com/vmware/govmomi/issues/1200
 	soapClient := soap.NewClient(u, dc.AllowInsecure)
 	// set our CA bundle
-	soapClient.DefaultTransport().TLSClientConfig = &tls.Config{
-		RootCAs: caBundle,
-	}
+	soapClient.DefaultTransport().TLSClientConfig.RootCAs = caBundle
```
Comment (xrstf) [inline pkg/provider/cloud/vsphere/provider.go:93]: Are we sure `TLSClientConfig` is never nil?

**Review by xrstf — COMMENTED** [review]
Q

File: pkg/provider/cloud/vsphere/provider.go (hunk 1 above)
Comment (mate4st) [reply pkg/provider/cloud/vsphere/provider.go:93]: It gets initialized in soap.NewClient():

`c.t.TLSClientConfig = &tls.Config{InsecureSkipVerify: c.k}`

**Review by kron4eg — APPROVED** [review]
/retest

**Review by xrstf — APPROVED** [review]
/approve

**Pull request merged**